### PR TITLE
Fix indeterminate z-order of EditedMediaItemSequences

### PR DESCRIPTION
**Pull request by AradiPatrik**
## Root Cause:
The indeterminate z-order issue arises due to the non-deterministic loading of assets. As each asset finishes loading, DefaultVideoCompositor.registerInput is invoked, linking the z-order to the unpredictable asset loading sequence. Consequently, the final z-order in the output is uncertain, as it's contingent on the load completion sequence of the assets.

## Proposed Fix:
To resolve this, I've explored two options:

- *Option A*: Modify TransformerInternal.java to aggregate all loaded assets first, then sequentially invoke registerInput in the intended order.
- *Option B*: Introduce a sequenceIndex parameter to registerInput, decoupling the z-order from the invocation sequence and aligning it with this index instead.

## Implemented Solution:
I opted for Option B due to its straightforward implementation. Post-fix, the z-order issue was no longer reproducible, indicating the effectiveness of this solution.

## Links
- [Original issue](https://github.com/androidx/media/issues/1029)
- [Simple Sample Project for reliable reproduction](https://github.com/AradiPatrik/Media3Issue)

I'm open to feedback and willing to assist further to ensure a robust resolution to this z-order inconsistency.

### CODE REVIEW

**Comment by dway123** [comment]
Thank you for the quick PR! I'll try to take a look in the next few days. Also, cc @claincly who is pretty familiar with these files too

**Comment by dway123** [comment]
I have less familiarity than @claincly here so I'll defer to him, but just noting that without as much context, I think the general option B selected in this PR sounds good to me (haven't looked to make sure the implementation style/architecture makes sense though). Option A would also require adding additional state and/or blocking, so I'd prefer B over A.

File: libraries/effect/src/main/java/androidx/media3/effect/VideoCompositor.java
Comment (claincly) [inline libraries/effect/src/main/java/androidx/media3/effect/VideoCompositor.java:55]: a high level comment: it's strange that this method takes in an ID and returns another ID, where those IDs can potentially be the same. In the actual impl(in `DefaultVideoCompositor`) those two indices match and that's what fixes the z-order problem. 

I feel the approach could be a bit top down, that this method just takes a `sequenceIndex` but doesn't return any index. The passed-in index ultimately defines the z-order, wdyt? @dway123 please also take a look and see if you agree.

Comment (AradiPatrik) [reply libraries/effect/src/main/java/androidx/media3/effect/VideoCompositor.java:55]: @claincly This approach sounds good to me!

Let me change the code according to this design.

Comment (AradiPatrik) [reply libraries/effect/src/main/java/androidx/media3/effect/VideoCompositor.java]: @dway123 @claincly Implemented the top down approach for id passing as discussed.

File: libraries/effect/src/main/java/androidx/media3/effect/MultipleInputVideoGraph.java
Comment (dway123) [inline libraries/effect/src/main/java/androidx/media3/effect/MultipleInputVideoGraph.java:137]: isn't isEmpty equivalent to size() == 0?

File: libraries/transformer/src/main/java/androidx/media3/transformer/AudioSampleExporter.java
Comment (dway123) [inline libraries/transformer/src/main/java/androidx/media3/transformer/AudioSampleExporter.java:97]: @claincly  not sure but does Audio also need this information?

**Review by dway123 — COMMENTED** [review]
(some small comments at a glance :P )

File: libraries/effect/src/main/java/androidx/media3/effect/VideoCompositor.java
Comment (dway123) [reply libraries/effect/src/main/java/androidx/media3/effect/VideoCompositor.java]: @claincly ptal?

File: libraries/effect/src/main/java/androidx/media3/effect/MultipleInputVideoGraph.java
Comment (claincly) [inline libraries/effect/src/main/java/androidx/media3/effect/MultipleInputVideoGraph.java:262]: nit: Can use `Util.contains(preProcessors, inputId)` here.

File: libraries/common/src/main/java/androidx/media3/common/VideoGraph.java
Comment (claincly) [inline libraries/common/src/main/java/androidx/media3/common/VideoGraph.java:78]: It seems it's easier if we require sequence indices start from 0, so please add something like "The index must start from zero"

File: libraries/effect/src/main/java/androidx/media3/effect/SingleInputVideoGraph.java
Comment (claincly) [inline libraries/effect/src/main/java/androidx/media3/effect/SingleInputVideoGraph.java:41]: nit: `The index of the only {@linkplain #registerInput(int) registered} input.`

File: libraries/effect/src/main/java/androidx/media3/effect/VideoCompositor.java
Comment (claincly) [inline libraries/effect/src/main/java/androidx/media3/effect/VideoCompositor.java:52]: Please reword: `which is used to determine the order of the input sources`

File: libraries/transformer/src/main/java/androidx/media3/transformer/AudioSampleExporter.java
Comment (claincly) [reply libraries/transformer/src/main/java/androidx/media3/transformer/AudioSampleExporter.java:97]: I'm not sure either, but I think it's OK since it's from the `SampleExporter` API change.

File: libraries/transformer/src/main/java/androidx/media3/transformer/SampleExporter.java
Comment (claincly) [inline libraries/transformer/src/main/java/androidx/media3/transformer/SampleExporter.java:67]: supernit: `The index of the input sequence`?

**Review by claincly — CHANGES_REQUESTED** [review]
Look good overall!

**Comment by AradiPatrik** [comment]
@claincly I implemented the changes you requested. Please take a look and resolve it if it's OK. Let me know if there's anything else we need to change before this PR can be merged.

**Comment by claincly** [comment]
I'm going to send this for internal review now. You may see some more commits being added as I make changes in response to review feedback. Please refrain from pushing any more substantive changes as it will complicate the internal review - thanks!

**Comment by AradiPatrik** [comment]
@claincly Any updates on this? Can I assist in any way?

**Comment by claincly** [comment]
Hey Aradi, we are discussing it internally as it's an API change, will keep you in the loop!

**Comment by yuriikonovalov** [comment]
@claincly could you provide some information (if it's possible) when it's potentially can be merged and released so that this fix is available to use in production?

**Pull request merged**
